src/main/java/ama/post/CommentPost.java

```diff
@@ -29,12 +38,29 @@ public Post getContext() {
         return this.context;
     }
 
-    /*  don't think we need this anymore
-    public Post getSubmissionPost() {
-        if(this.context instanceof SubmissionPost) {
-            return this.context;
-        } else {
-            return ((CommentPost)this.context).getSubmissionPost();
+    public int getUpvotes() {
+        return this.upvoters.size();
+    }
+
+    public int getDownvotes() {
+        return this.downvoters.size();
+    }
+
+    public void upvote(User user) {
```

Comment:
Nice. It makes it so that if you previously downvoted, and now you wish to upvote, it will switch it so that the user can't bot like and dislike the comment. Good stuff :+1: